vyper/builtin_functions/signatures.py

```diff
@@ -101,43 +59,24 @@ def validate_inputs(wrapped_fn):
 
     @functools.wraps(wrapped_fn)
     def decorator_fn(self, node, context):
-        argz = [i[1] for i in self._inputs]
-        kwargz = getattr(self, "_kwargs", {})
-        function_name = node.func.id
-        if len(node.args) > len(argz):
-            raise StructureException(
-                f"Expected {len(argz)} arguments for {function_name}, got {len(node.args)}", node
-            )
         subs = []
-        for i, expected_arg in enumerate(argz):
-            if len(node.args) > i:
-                subs.append(
-                    process_arg(
-                        i + 1,
-                        node.args[i],
-                        expected_arg,
-                        function_name,
-                        context,
-                    )
-                )
-            elif isinstance(expected_arg, Optional):
-                subs.append(expected_arg.default)
-            else:
-                raise StructureException(f"Not enough arguments for function: {node.func.id}", node)
+        for arg in node.args:
+            subs.append(process_arg(arg, arg._metadata["type"], context))
```

Comment:
nit: since `process_arg` has side effects, break it out onto its own line
```suggestion
            arg_ir = process_arg(arg, arg._metadata["type"], context)
            subs.append(arg_ir)
```